pkg/secrets/strategy.go

```diff
@@ -130,18 +118,128 @@ func (s Source) MarshalYAML() (interface{}, error) {
 	return s.MarshalRaw(), nil
 }
 
-type StrategyList []SourceMap
+// SourceMapList is a list of mappings that can be access via index or the item name.
+type SourceMapList []SourceMap
 
-func (l StrategyList) Less(i, j int) bool {
+func (l SourceMapList) Less(i, j int) bool {
 	return l[i].Name < l[j].Name
 }
 
-func (l StrategyList) Swap(i, j int) {
+func (l SourceMapList) Swap(i, j int) {
 	tmp := l[i]
 	l[i] = l[j]
 	l[j] = tmp
 }
 
-func (l StrategyList) Len() int {
+func (l SourceMapList) Len() int {
 	return len(l)
 }
+
+// HasName determines if the specified name is defined in the set.
+func (l SourceMapList) HasName(name string) bool {
+	_, ok := l.GetByName(name)
+	return ok
+}
+
+// GetByName returns the resolution strategy for the specified name and a bool indicating if it was found.
+func (l SourceMapList) GetByName(name string) (SourceMap, bool) {
```

Comment:
would it be worth returning an `error` with a message like `fmt.Errorf("SourceMapList has no name %s", name)`?

Reply:
yea, I agree it shoudn't be a `(SourceMap, bool, error)` return. I wasn't clear enough before, sorry. I was asking if you think we could do a `(SourceMap, error)` return value, and then if the name wasn't found, you could return a `fmt.Errorf(...)` in that case, rather than `false`?